test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/ODataUriParserTests.cs

```diff
@@ -26,6 +26,410 @@ public class ODataUriParserTests
+            container.AddEntitySet("foos", foo);
+            model.AddElement(container);
+
+            var parser = new ODataUriParser(model, new Uri("/foos?$select=NaN", UriKind.Relative));
+            var odataUri = parser.ParseUri();
+
+            var selectedItems = odataUri.SelectAndExpand.SelectedItems.ToArray();
+            Assert.Single(selectedItems);
+            var pathSelectItem = selectedItems[0] as PathSelectItem;
+            var segments = pathSelectItem.SelectedPath.Segments.ToArray();
+            Assert.Single(segments);
+            Assert.Equal("NaN", segments[0].Identifier);
+            Assert.Equal("Edm.String", segments[0].EdmType.FullTypeName());
+        }
+
+        /// <summary>
+        /// Selects a property named "null"
+        /// </summary>
+        [Fact]
+        public void SelectNull()
+        {
+            var model = new EdmModel();
+            var foo = new EdmEntityType("foobar", "foo");
+            foo.AddKeys(foo.AddStructuralProperty("id", EdmCoreModel.Instance.GetInt32(false)));
+            foo.AddStructuralProperty("null", EdmCoreModel.Instance.GetString(true));
+            model.AddElement(foo);
+            var container = new EdmEntityContainer("foobar", "FooService");
+            container.AddEntitySet("foos", foo);
+            model.AddElement(container);
+
+            var parser = new ODataUriParser(model, new Uri("/foos?$select=null", UriKind.Relative));
+            var odataUri = parser.ParseUri();
+
+            var selectedItems = odataUri.SelectAndExpand.SelectedItems.ToArray();
+            Assert.Single(selectedItems);
+            var pathSelectItem = selectedItems[0] as PathSelectItem;
+            var segments = pathSelectItem.SelectedPath.Segments.ToArray();
+            Assert.Single(segments);
+            Assert.Equal("null", segments[0].Identifier);
+            Assert.Equal("Edm.String", segments[0].EdmType.FullTypeName());
+        }
+
+        /// <summary>
+        /// Selects a property named "null" when such a property doesn't exist on the entity
+        /// </summary>
+        [Fact]
+        public void SelectNullNotFound()
+        {
+            var model = new EdmModel();
+            var foo = new EdmEntityType("foobar", "foo");
+            foo.AddKeys(foo.AddStructuralProperty("id", EdmCoreModel.Instance.GetInt32(false)));
+            foo.AddStructuralProperty("true", EdmCoreModel.Instance.GetString(true));
+            model.AddElement(foo);
+            var container = new EdmEntityContainer("foobar", "FooService");
+            container.AddEntitySet("foos", foo);
+            model.AddElement(container);
+
+            var parser = new ODataUriParser(model, new Uri("/foos?$select=null", UriKind.Relative));
+            var odataException = Assert.Throws<ODataException>(() => parser.ParseUri());
+
+            Assert.Equal("Could not find a property named 'null' on type 'foobar.foo'.", odataException.Message);

```

Comment:
Should have another test where the type is defined as open -- in which case, no error should be returned.